modules/howtos/pages/vector-searching-with-sdk.adoc

```diff
@@ -0,0 +1,216 @@
+This can be more efficient, unless you are doing a lot of optimization to your query.
+
+.Parameterized Vector Query
+[source,python]
+----
+  result = cluster.query(
+         "SELECT d.id, d.question, d.wanted_similar_color_from_search, " +
+         "  ARRAY_CONCAT( " +
+           "d.couchbase_search_query.knn[0].vector[0:4], " +
+           "['...'] " +
+         ") AS vector " +
+      "FROM `vector-sample`.`color`.`rgb-questions` AS d " +
+  	"WHERE d.id = $id;",
+  queryOptions()
+  	.parameters = {"id": "#87CEEB"}
+
+    for row in result.rows():
+        print(f"Found match: {row}")
+----
+
+
+== Vector Search With the Search Service
+
+Vector search is also implemented using xref:full-text-searching-with-sdk.adoc[Search Indexes], and can be combined with traditional full text search queries.
+Vector embeddings can be an array of floats or a xref:server:vector-search:run-vector-search-ui.adoc#base64[base64 encoded string].
+
+=== Prerequisites
+
+Couchbase Server 7.6.0
+(7.6.2 for base64-encoded vectors) -- or a Capella Operational cluster.
+
+=== Examples
+
+==== Single vector query
+
+In this first example we are performing a single vector query:
+
+[source,python]
+----
+include::howtos:example$search_ops.py[tag=vector_search_single]
+----
+
+Let's break this down.
+We create a `SearchRequest`, which can contain a traditional FTS query `SearchQuery` and/or the new `VectorSearch`.
+Here we are just using the latter.
+
+The `VectorSearch` allows us to perform one or more `VectorQuery` s.
+
+The `VectorQuery` itself takes the name of the document field that contains embedded vectors ("vector_field" here), plus actual vector query in the form of a `float[]`.
+
+(Note that Couchbase itself is not involved in generating the vectors, and these will come from an external source such as an embeddings API.)
+
+Finally we execute the `SearchRequest` against the FTS index "travel-sample-index", which has previously been setup to vector index the "vector_field" field.
+
+This happens to be a scoped index so we are using `scope.search()`.
+If it was a global index we would use `cluster.search()` instead -- see <<Scoped vs Global Indexes>>.
+
+It returns the same `SearchResult` detailed on the xref:full-text-searching-with-sdk.adoc[Search page].
+
+
+==== Pre-Filters
+
+From Couchbase Server 7.6.4 -- and in Capella Operational clusters -- 
+xref:server:vector-search:pre-filtering-vector-search.adoc#about-pre-filtering[pre-filtering with similarity search] is available.
+This is a non-vector query that the server executes first to get an intermediate result.
+Then it executes the vector query on the intermediate result to get the final result.
+
+[source,python]
+----
+VectorQuery(field_name, vector, num_candidates=None, boost=None, prefilter=SearchQuery)
```

Comment:
I think it might be best to provide a specific `SearchQuery`?  Unless we are wanting to give the signature of `VectorQuery`?

If providing a concrete example for creating a `VectorQuery` with a prefilter, I think something like below would be ideal.

```
VectorQuery.create(field_name, vector, prefilter=MatchQuery('primary', field='color_wheel_pos'))
```

If wanting to show the signature, it is:
```
    def __init__(self,
                        field_name: str,
                        vector: Union[List[float], str],
                        num_candidates: Optional[int] = None,
                        boost: Optional[float] = None,
                        prefilter: Optional[SearchQuery] = None
                        ) -> None:
```